src/docs/testing/common-errors.md

```diff
@@ -0,0 +1,451 @@
+
+<!-- ![alt_text](images/image1.png "image_tooltip") -->
+
+
+```
+The following assertion was thrown during layout:
+A RenderFlex overflowed by 1146 pixels on the right.
+
+The relevant error-causing widget was
+    Row 							lib/errors/renderflex_overflow_column.dart:23
+The overflowing RenderFlex has an orientation of Axis.horizontal.
+The edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow and black striped pattern. This is usually caused by the contents being too big for the RenderFlex.
+
+Consider applying a flex factor (e.g. using an Expanded widget) to force the children of the RenderFlex to fit within the available space instead of being sized to their natural size.
+This is considered an error condition because it indicates that there is content that cannot be seen. If the content is legitimately bigger than the available space, consider clipping it with a ClipRect widget before putting it in the flex, or using a scrollable container rather than a Flex, like a ListView.
+
+The specific RenderFlex in question is: RenderFlex#c3a70 relayoutBoundary=up1 OVERFLOWING
+```
+
+
+
+### How might you run into this error?
+
+The error often occurs when a Column or Row has a child widget that is not constrained in its size. Let’s take a look at a common scenario in the code below:
+
+
+```dart
+Widget build(BuildContext context) {
+  return Container(
+    child: Row(
+      children: [
+        Icon(Icons.message),
+        Column(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Text("Title", style: Theme.of(context).textTheme.headline4),
+            Text(
+                "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed"
+                " do eiusmod tempor incididunt ut labore et dolore magna "
+                "aliqua. Ut enim ad minim veniam, quis nostrud "
+                "exercitation ullamco laboris nisi ut aliquip ex ea "
+                "commodo consequat."),
+          ],
+        ),
+      ],
+    ),
+  );
+}
+```
+
+
+In the above example, the Column will try to be wider than the space the Row (its parent) can allocate to it, causing an overflow error.  Why does the Column try to do that? To understand this layout behavior, we need to remember how Flutter Framework does layout:
+
+<!-- TODO: Fix the blockquote style -->
+>"To perform layout, Flutter walks the render tree in a depth-first traversal and **passes down size constraints** from parent to child… Children respond by **passing up a size** to their parent object within the constraints the parent established."  – [Flutter architectural overview](/docs/resources/architectural-overview#layout-and-rendering)
```

Comment:
@sfshaza2 @domesticmouse I can't get the blockquote style to work in my local build. Do you know the right way to do it?

Reply:
I'm looking for the Github-like quote effect:

>To perform layout, Flutter walks the render tree in a depth-first traversal and **passes down size constraints** from parent to child… Children respond by **passing up a size** to their parent object within the constraints the parent established.